### PR TITLE
rpc: modify VerifyChanBackup to take either a Single or Multi

### PR DESCRIPTION
**Pull request by wpaulino**

### CODE REVIEW

File: lnrpc/rpc.proto [hunk 1]
```diff
@@ -733,10 +733,8 @@ service Lightning {
     };
 
     /**
-    VerifyChanBackup allows a caller to verify the integrity of a channel
-    backup snapshot. This method will accept both a packed Single, and also a
-    Packed multi. Two bools are returned which indicate if the passed Single
-    (if present) is valid and also if the passed Multi (if present) is valid.
+    VerifyChanBackup allows a caller to verify the integrity of a channel backup
+    snapshot. This method will accept either a packed Single or a Packed multi.
```
Comment (cfromknecht) [inline lnrpc/rpc.proto]: comment should mention that specifying both will return an error

File: rpcserver.go [hunk 2]
```diff
@@ -4766,63 +4766,61 @@ func (r *rpcServer) ExportChannelBackup(ctx context.Context,
...
 			"backup must be specified")
-	}
 
-	// If a Single is specified then we'll only accept one of them to allow
-	// the caller to map the valid/invalid state for each individual
-	// Single.
-	if in.GetSingleChanBackups() != nil &&
-		len(in.GetSingleChanBackups().ChanBackups) != 1 {
+	// Either a Single or a Multi must be specified, but not both.
+	case in.GetSingleChanBackups() == nil && in.GetMultiChanBackup() == nil:
```
Comment (cfromknecht) [inline rpcserver.go]: identical to first case

File: rpcserver.go [hunk 3]
```diff
@@ -4766,63 +4766,61 @@ func (r *rpcServer) ExportChannelBackup(ctx context.Context,
...
 		packedMultiBackup := in.GetMultiChanBackup().MultiChanBackup
 		packedMulti := chanbackup.PackedMulti(packedMultiBackup)
 
-		// We'll now attempt to unpack the Multi. If this fails, then
-		// we know it's invalid.
+		// We'll now attempt to unpack the Multi. If this fails, then we
+		// know it's invalid.
 		_, err := packedMulti.Unpack(r.server.cc.keyRing)
-		resp.MultiValid = err == nil
+		valid = err == nil
```
Comment (cfromknecht) [inline rpcserver.go]: might be better to just return the error directly, and the user can infer that it's valid if there was no error

Comment (wpaulino) [reply rpcserver.go]: Fixed.

File: rpcserver.go (hunk 2 above)
Comment (wpaulino) [reply rpcserver.go]: Fixed.

File: lnrpc/rpc.proto (hunk 1 above)
Comment (wpaulino) [reply lnrpc/rpc.proto]: Fixed.

**Comment by Roasbeef** [comment]
@joostjager the separation allows distinct packing/encoding formats for a Single vs a Multi. A raw Single is also smaller than a Multi with only one channel.

**Pull request merged**